### ISSUE TITLE
BUG: Docs website is blocked by google safe browsing

### ISSUE THREAD

**Issue by nylocx** [pandas-dev/pandas#39862]
I hope it's ok to skip the template as this is not a direct code bug but a website bug.
Sorry for the german locales, but I guess the message is pretty clear anyways. The pandas docs are blocked by google safe browsing service.

![image](https://user-images.githubusercontent.com/3329124/108188866-64d49180-7110-11eb-8b42-b67adddd4db3.png)

Hope this can be clarified quickly as there will be a a lot of support lines in companies running hot ;).

**Comment by attack68** [comment]
I had the same error as you but recently the status seems to been updated by Google:
![Screen Shot 2021-02-17 at 13 31 28](https://user-images.githubusercontent.com/24256554/108204813-8b9cc300-7124-11eb-8e0a-c6c36d240cc2.png)
This was showing unsafe earlier. Perhaps it will feed through to browsers shortly...??

**Comment by attack68** [comment]
Yeh has error:
![Screen Shot 2021-02-17 at 13 33 13](https://user-images.githubusercontent.com/24256554/108204969-bf77e880-7124-11eb-8b76-e6c74c11e87a.png)

**Comment by jorisvandenbossche** [comment]
Yes, I get the same. The website itself (https://pandas.pydata.org/) is still working, as well as other documentation links (eg stable at https://pandas.pydata.org/pandas-docs/stable/, dev version at https://pandas.pydata.org/pandas-docs/dev/).  
So it only seems to be the case for the `/doc/` urls

The status says (https://transparencyreport.google.com/safe-browsing/search?url=https:%2F%2Fpandas.pydata.org%2Fdocs%2F): *"Install unwanted or malicious software on visitors’ computers"*

cc @pandas-dev/pandas-core @pandas-dev/pandas-triage anybody with more web knowledge that knows how to investigate this problem?

**Comment by toobaz** [comment]
Strangely, nothing malicious is reported in the "transparency report" by Google as of now...

https://transparencyreport.google.com/safe-browsing/search?url=https:%2F%2Fpandas.pydata.org

**Comment by datapythonista** [comment]
Difficult to guess without Google providing any information on why the block us.

I'd assume it's some javascript stuff in the page. Looks like there is a Jupyter widget that feels it could be the problem. Also, looks like jquery version was bumped recently in the theme, but I don't think something as popular as jquery should be it.

I'd ssh to the server, and comment some of those `<script>` blocks one by one, and hopefully one of them makes the page immediately stop being blocked (if Google is using a database and changes are not immediate, then I don't know what can be done to debug).

**Comment by toobaz** [comment]
> Strangely, nothing malicious is reported in the "transparency report" by Google as of now...

Oh wait... ``https://pandas.pydata.org/docs/`` is problematic while ``https://pandas.pydata.org/docs`` is not!

**Comment by fjetter** [comment]
https://transparencyreport.google.com/safe-browsing/search?url=pandas.pydata.org%2Fdocs%2F&hl=en-us
![image](https://user-images.githubusercontent.com/8629629/108215681-e25cc980-7131-11eb-8b76-1efe9901258a.png)

unfortunately, there is nothing else on that page

**Comment by jorisvandenbossche** [comment]
The overview page on that transparencyreport google website has a section **"How we notify webmasters"**: https://transparencyreport.google.com/safe-browsing/overview

Where they say they alert the webmaster of the site. But I don't know who whould be contacted in our case. 
There are also some links in that section on how to get more details. But it seems you need to register and verify that you are owner of the website in the "Google Search Console"

**Comment by toobaz** [comment]
> But it seems you need to register and verify that you are owner of the website in the "Google Search Console"

Yes, I think this the way to go, they should ask you to put a token in the DNS records to prove ownership.

**Comment by toobaz** [comment]
> Yes, I think this the way to go, they should ask you to put a token in the DNS records to prove ownership.

(alternatively, it is possible to prove ownership of the URL prefix by uploading some file/html code)

**Comment by toobaz** [comment]
I made a test, and if we insert the tag

``<meta name="google-site-verification" content="jiLZBCawTaPk1EB8jeQxmZoDSsTbPDcdcKPpUvj-HBw" />``

... in the html, this should suffice for me to claim control. Not that I'm necessarily the most qualified to claim control, obviously.

**Comment by jorisvandenbossche** [comment]
It might be working again? (it seems this is the case for me) Although the report still says it is unsafe (https://transparencyreport.google.com/safe-browsing/search?url=https:%2F%2Fpandas.pydata.org%2Fdocs%2F)

**Comment by datapythonista** [comment]
Still failing for me on Brave.

**Comment by jorisvandenbossche** [comment]
Yeah, now again getting the failure on chrome as well (for some reason 10 min ago it didn't fail). The chrome message gives a bit more explicit link about how to fix it: https://developers.google.com/web/fundamentals/security/hacked/request_review (but also pointing to verifying ownershop with Google Search Console)

**Comment by Dr-Irv** [comment]
Joris:
It could be a cache issue.  For me, it was fine on Chrome, but in Firefox,
it reported it as unsafe.  For what it's worth, Microsoft Edge has no
issues, but they probably don't use the Google scanner!

-Irv

**Comment by martey** [comment]
Since pandas.pydata.org is a subdomain of pydata.org, people with Google Search Console access to that domain were notified.

The file that Google says is causing problems is https://pandas.pydata.org/docs/pandas.zip, the zipped version of the documentation, so I can't tell whether the issue is a false positive based on the entire ZIP file, or a specific issue with an individual file in the documentation site.

If there haven't been any recent changes to the documentation that might be causing this, I can request a review through the Search Console. It would be good if someone with more experience with pandas documentation could check things beforehand - in case there is an actual issue or to help determine whether there is a more specific file that Google is flagging.

**Comment by martey** [comment]
I'm not seeing any recent changes that could have caused this, and reviewing the ZIP file did not show any obvious problem, so I am believe that this is a false positive. I have requested a review in the Search Console.

**Comment by kaymal** [comment]
I still have the problem on Chrome, with the following message.
> Attackers on pandas.pydata.org might attempt to trick you into installing programs that harm your browsing experience (for example, by changing your homepage or showing extra ads on sites you visit)

**Comment by bede** [comment]
Still an issue using Firefox

**Comment by vincentdavis** [comment]
virustotal shows google safe browsing still reporting but no others.
https://www.virustotal.com/gui/url/4fe68634bce91a0bb647919f9728e86994e5c8c61a2039b18ec8349d32636c28/detection

**Comment by vincentdavis** [comment]
Sorry if I am just noise.
When I do a google search for ```pandas concat```, I get this url which has the warning.
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.concat.html

If I go to pandas.pydata.org the make my way to the similar page I get no warning and the url is:
https://pandas.pydata.org/docs/reference/api/pandas.concat.html?highlight=concat

pydata.org gets a warning that some "Some pages on this site are unsafe"
https://transparencyreport.google.com/safe-browsing/search?url=pydata.org

**Comment by topper-123** [comment]
I unpinned this by mistake and am on my phone. Can you pin it again, @MarcoGorelli ?

**Comment by beojan** [comment]
Now, "stable" seems to be blocked while https://pandas.pydata.org/docs/ is fine

**Comment by philippefutureboy** [comment]
For me issue was not there yesterday but is there today. Idk if that info can be of help.
In the meanwhile, is the website safe to browse? I need my pandas fix 🤪

**Comment by vincentdavis** [comment]
Un-report a website.

https://developers.google.com/safe-browsing/v4/reporting

**Comment by MohammedMUddin** [comment]
http://pandas.pydata.org/pandas-docs/stable/io.html#io-store-in-csv 
Getting the warning for this link.

**Comment by rth** [comment]
> Where they say they alert the webmaster of the site. But I don't know who would be contacted in our case.

Probably the owners of pydata.org cf message by @aterrel on the mailing list.

**Comment by aterrel** [comment]
Can ya'll remove the zip file that @martey pointed out?

**Comment by TomAugspurger** [comment]
Good idea. I removed that file (I downloaded it locally if needed).

**Comment by Tayflo** [comment]
It seems fixed everywhere on my side, on Firefox (also tried quickly Chrome and Edge).

Plus:

- https://transparencyreport.google.com/safe-browsing/search?url=pydata.org
- https://transparencyreport.google.com/safe-browsing/search?url=https:%2F%2Fpandas.pydata.org
- https://transparencyreport.google.com/safe-browsing/search?url=https:%2F%2Fpandas.pydata.org%2Fdocs%2F
- https://transparencyreport.google.com/safe-browsing/search?url=https:%2F%2Fpandas.pydata.org%2Fpandas-docs%2Fstable%2F

saying all "✅ No unsafe content"

Virus total not up-to-date tho: https://www.virustotal.com/gui/url/4fe68634bce91a0bb647919f9728e86994e5c8c61a2039b18ec8349d32636c28/detection

I got a notification email from this GitHub conversation in Gmail, flagged unsafe (something like "links to website containing malicious software..."):

> @beojan
> Now, "stable" seems to be blocked while https://pandas.pydata.org/docs/ is fine

But it was 16 hours ago, it seems ok now (I tried to send me a mail with the link from one address to another).

**Comment by aterrel** [comment]
@TomAugspurger thanks. you might see if you can generate the zip file on your computer and compare the md5 hash of the two. If they are compromised this might be a good time to just migrate that server over to the pandas AWS account as the Rackspace account will be deprecated at the end of 2021.

@Tayflo I'm also seeing things stable. Hopefully this will resolve in the next 48 hours.

**Comment by AveCoders** [comment]
I've recently done a tutorial on Pandas and it was blocked by YouTube with the appeal been rejected on the grounds that I'm violating community guidelines by producing harmful content. Any suggestions?

**Comment by MarcoGorelli** [comment]
@AveCoders please ask on Gitter or open a new issue, there's no need to keep commenting on unrelated issues

**Comment by rhshadrach** [comment]
I haven't been having any issues in at least the past two days - are others experiencing the same?

**Comment by philippefutureboy** [comment]
I confirm that issue is solved on my side! Congrats to the team! 👏🎉

**Comment by afeld** [comment]
Seems this is resolved, so closing. Thanks everyone!

**Comment by jorisvandenbossche** [comment]
@TomAugspurger were you able to check the content of the zip file?

**Comment by simonjayhawkins** [comment]
i've still got the original file if that helps

```
(base) simon@DESKTOP-HJVF5J5:~/pandas-release/pandas (1.2.x)$ sha256sum doc/build/html/pandas.zip                       
8f85a4e5317e827ee4e92e616968bd7395c02087418e4ed46992223a7259fec5  doc/build/html/pandas.zip   
```

**Comment by TomAugspurger** [comment]
I haven't had a chance to look.

Do we try to rebuild the zipped HTML and re-upload it? https://pandas.pydata.org/docs/pandas.zip is 404ing right now.

**Comment by martey** [comment]
I think it would be prudent to determine whether the rebuilt zipped HTML is exactly the same as the one that was previously on the website, as @aterrel suggested earlier.

If the checksum of the local build differs, it could mean the server was compromised.

If the checksum is the same, it is likely a false positive, but re-uploading will likely cause the original problem to start happening again.

**Comment by TomAugspurger** [comment]
I don't know if Sphinx generates byte-for-byte identical outputs, but it's worth a shot. Anyone have time to attempt that? The scripts are in https://github.com/pandas-dev/pandas-release.

**Comment by jorisvandenbossche** [comment]
@TomAugspurger Simon posted the hash of the original zip file he uploaded a few comments above. And I think you have downloaded the file from the server that caused the problems? So I think if you could check the hash of that zip file, and compare with the one above, that would be a good start.

**Comment by TomAugspurger** [comment]
Good point:

```
~ » shasum -a 256 pandas.zip
8f85a4e5317e827ee4e92e616968bd7395c02087418e4ed46992223a7259fec5  pandas.zip

```

Which seems to match.

**Comment by john-sandall** [comment]
I'm still seeing this on Twitter links:

![image](https://user-images.githubusercontent.com/2884159/109424290-01cddf00-79db-11eb-88ea-4b85ebeef15d.png)

There's a [link in this post](https://twitter.com/John_Sandall/status/1366044821509578760) that could be useful for testing this.

**Comment by aterrel** [comment]
@john-sandall looks like you can report miscaterogization of the link at https://help.twitter.com/en/forms/authenticity/spam

**Comment by MarcoGorelli** [comment]
Thanks @john-sandall for the report - reopening until this is fixed there then

**Comment by MartinThoma** [comment]
I don't get any warning anymore. Is the problem gone?

**Comment by rhshadrach** [comment]
Confirming no issues on my end since February.

**Comment by rhshadrach** [comment]
Twitter is still giving a warning. I reported the link to Twitter (wasn't clear if anyone else had), and will update.

**Comment by rhshadrach** [comment]
I got back a generic response, "Though we cannot respond to individual reports, the information provided helps us make Twitter better for everyone." Unless others also want to report it, I do not think there is anything more to be done here.

**Comment by MarcoGorelli** [comment]
cool, let's close then, thanks reporting it to them!
